pom.xml

```diff
@@ -1,242 +1,131 @@
                 <configuration>
-                  <executable>${java.home}/bin/java</executable>
-                  <arguments>
-                    <argument>-classpath</argument>
-                    <classpath/>
-                    <argument>-ea</argument>
-                    <argument>com.github.jbellis.jvector.example.SiftSmall</argument>
-                  </arguments>
+                    <release>11</release>
                 </configuration>
-              </execution>
-              <execution>
-                <id>bench</id>
-                <goals>
-                  <goal>exec</goal>
-                </goals>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>3.3.0</version>
+            </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.1.2</version>
                 <configuration>
-                  <executable>${java.home}/bin/java</executable>
-                  <arguments>
-                    <argument>-classpath</argument>
-                    <classpath/>
-                    <argument>-Xmx32G</argument>
-                    <argument>-ea</argument>
-                    <argument>com.github.jbellis.jvector.example.Bench</argument>
-                  </arguments>
+                    <skip>true</skip>
                 </configuration>
-              </execution>
-            </executions>
-          </plugin>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.junit.vintage</groupId>
+                        <artifactId>junit-vintage-engine</artifactId>
+                        <version>5.9.1</version>
+                    </dependency>
+                </dependencies>
+            </plugin>
         </plugins>
-      </build>
-    </profile>
-  </profiles>
+    </build>
+    <dependencies>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.9.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.carrotsearch.randomizedtesting</groupId>
+            <artifactId>randomizedtesting-runner</artifactId>
+            <version>2.8.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.openjdk.jmh</groupId>
+            <artifactId>jmh-core</artifactId>
+            <version>1.35</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.openjdk.jmh</groupId>
+            <artifactId>jmh-generator-annprocess</artifactId>
+            <version>1.35</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+    <modules>
+        <module>jvector-base</module>
```

Comment:
We are going to want to make an `examples` so we dont ship that code in the core jar, but that is out of scope for this issue.